### PR TITLE
Introduction of Dockerfiles for both local and prod env in frontend directory

### LINKED ISSUE #94: Introduce Docker Files for Local and Production Environments
## Description  
To streamline the development, testing, and deployment processes for the React + TypeScript + Vite frontend project, I propose introducing Docker files tailored for different stages:  
1. **Local Development**: A Docker setup optimized for local development with hot-reloading and minimal overhead.  
2. **Production**: A Docker setup optimized for production with a lightweight image and efficient build process.  

## Why This is Important  
- **Consistency Across Environments**: Docker ensures that the application runs consistently across different environments (local, staging, production).  
- **Simplified Onboarding**: New developers can quickly set up the project without worrying about local dependencies or configurations.  
- **Improved Developer Experience**: A local Docker setup with hot-reloading will make development faster and more efficient.  

## Proposed Approach  
1. **Local Development Dockerfile**:  
   - Use a base image like `node:alpine` for lightweight development.  
   - Include dependencies for Vite and React development.  
   - Enable hot-reloading for a seamless development experience.  

2. **Production Dockerfile**:  
   - Use a multi-stage build to create a lightweight production image.  
   - Build the Vite project in one stage and serve the static files using a lightweight web server like `nginx`.
   - Optimize the image size by excluding unnecessary files (e.g., `node_modules`, build cache).  

## Next Steps  
1. Discuss and finalize the requirements for the local and production Docker setups.  
2. Create the Docker files and test them for both local and production environments.  
3. Document the usage of Docker files in the project README for easy adoption by the team.  

## Example Directory Structure  
```plaintext
frontend/
├── Dockerfile.local      # For local development
├── Dockerfile.prod       # For production
├── docker-compose.yml    
├── src/                
├── public/              
└── package.json

### PR DESCRIPTION
**Pull request by yashpandey06**
## Description  

Fixes #94

1. **Local Development**: A Docker setup optimized for local development with hot-reloading and minimal overhead.  
2. **Production**: A Docker setup optimized for production with a lightweight image and efficient build process.  

##  Frontend directory structure
```plaintext
frontend/
├── Dockerfile.local    # For local development
├── Dockerfile.staging      # For production  
├── src/                
├── public/              
└── package.json

### CODE REVIEW

File: frontend/Dockerfile.dev
Comment (arkid15r) [inline frontend/Dockerfile.dev:1]: Could you use the existing convention (rename it to Dockerfile.local)

File: frontend/Dockerfile.prod
Comment (arkid15r) [inline frontend/Dockerfile.prod:1]: And this one to Dockerfile.staging (we don't have prod env yet)

**Review by arkid15r — COMMENTED** [review]
The missing part here is a modified `docker-compose.yaml` -- I'd expect it to be part of this PR to in order to test the new stack that includes the new frontend image/container.

**Comment by arkid15r** [comment]
Please also run pre-commit

File: frontend/Dockerfile.dev [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+WORKDIR /app
+
+COPY package*.json ./
+
+RUN npm install
+
+COPY . .
+
+EXPOSE 5173
```
Comment (harsh3dev) [inline frontend/Dockerfile.dev]: We are running the React Frontend on `PORT 3000` so it would be better to change `5173` to `3000`

**Comment by yashpandey06** [comment]
@arkid15r , the changes are fine i guess ? whats your view I have also verified the compose files working on local too

**Review by arkid15r — COMMENTED** [review]
>@arkid15r , the changes are fine i guess ? whats your view I have also verified the compose files working on local too

I've updated the local file -- we mount a volume with the code and run a command withing docker compose. Please check if this setup works for you now.

As for staging the situation is a bit different -- we want to use there Nest application as closest to our future production version as possible. This means that we need to use a separate docker image built with `dist` output optimized for real use environment (see the similar approach for staging backend -- `image: arkid15r/owasp-nest-backend:latest`). At this point I don't have a full understanding of how to implement it yet so you can either remove or keep staging files -- up to you.

File: docker-compose-staging.yaml
Comment (arkid15r) [inline docker-compose-staging.yaml:1]: Let's revert this file changes for now -- I'll take care of it when the frontend staging image is ready for deploy.

Comment (yashpandey06) [reply docker-compose-staging.yaml:1]: Done , Reverted the changes

**Review by arkid15r — APPROVED** [review]
LGTM 👍

**Pull request merged**